trpcultivate_genotypes/tests/src/Functional/GenotypesLoader/GenotypesLoaderBasePluginTest.php

```diff
@@ -154,5 +157,100 @@ public function testGenotypesLoaderPluginBase(){
     // Get sample filepath
     $grabbed_sample_file_path = $plugin->getSampleFilepath();
     $this->assertEquals($sample_file_path, $grabbed_sample_file_path, "The sample filepath grabbed by the getter method does not match.");
+
+    /****************************************************************************
+     *  TESTS for method getRecordPkey()
+     ****************************************************************************/
+    // Get the primary key of the organism we created using getRecordPkey()
+    $organism_id = $plugin->getRecordPkey('Organism', 'organism', 0, array(
+      'genus' => $genus,
+      'species' => $species
+    ));
+    $this->assertEquals($organism_id, $grabbed_organism_id, "The organism Id grabbed by getRecordPkey() does not match.");
+
+    // Try to select a record using an invalid mode
+    $exception_caught = FALSE;
+    try {
+      $organism_id = $plugin->getRecordPkey('Organism', 'organism', 5, array(
+        'genus' => $genus,
+        'species' => $species
+      ));
+    }
+    catch ( \Exception $e ) { 
+      $exception_caught = TRUE;
+    }
+    $this->assertTrue($exception_caught, "Did not catch exception for using an inavlid mode (5).");
+
+    // Try to insert a record that already exists and catch the expected exception
+    $exception_caught = FALSE;
+    try {
+      $organism_id = $plugin->getRecordPkey('Organism', 'organism', 1, array(
+        'genus' => $genus,
+        'species' => $species
+      ));
+    } 
+    catch ( \Exception $e ) { 
+      $exception_caught = TRUE;
+    }
+    $this->assertTrue($exception_caught, "Did not catch exception for trying to insert a duplicate.");
+
+    // Enter a duplicate record to trigger an exception
+    $organism_id = $connection->insert('1:organism')
```

Comment:
```suggestion
    $second_organism_id = $connection->insert('1:organism')
```

Reply:
Addressed in commit [336b7ba](https://github.com/TripalCultivate/TripalCultivate-Genetics/pull/20/commits/336b7ba8d341204ca2c084dcfde51563a4c06a0f)